pkg/ingester/ingester.go

```diff
@@ -1059,10 +1081,21 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 
 				return wrapWithUser(err, userID)
 			}
+			numNativeHistograms := len(ts.Histograms)
+			if numNativeHistograms > 0 {
+				hasNativeHistograms = true
+				lastNativeHistogram := ts.Histograms[numNativeHistograms-1]
+				for _, span := range lastNativeHistogram.PositiveSpans {
+					numNativeHistogramBuckets += int(span.Length)
+				}
+				for _, span := range lastNativeHistogram.NegativeSpans {
+					numNativeHistogramBuckets += int(span.Length)
+				}
+			}
```

Comment:
Is it valid for there to be both histograms and float samples in a single `TimeSeries` message? If so, do we need to do something here to only count the histogram if its timestamp is after the last float sample's timestamp?

Reply:
It is valid, but very rare.

Another question is whether the last histogram is the latest sample. In general it will be except when client sends out of order histograms.
In general since the active series metrics are scraped at some interval, any number is just a sample taken at a particular timestamp, so statistically it should not matter which one we take. Taking the first sample is marginally faster actually, since doesn't involve a mul+add :)

Reply:
I added a check to make sure the last histogram's timestamp is after the last float's timestamp, just to make sure the metric is actually a histogram at the latest sample.